Add documentation on the OTel Spring Boot starter (#3377)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/java/springboot.md

Reviewer: I am not 100% sure yet where in the docs this should live, @open-telemetry/java-approvers, is this "automatic instrumentation" (and could live under Automatic) or is this something different and we need to find a new home for it?

Reviewer 2: the spring boot starter is a couple of things:
* autoconfiguration of the OpenTelemetry SDK
* autoinstrumentation of a very limited set of libraries (logging and jdbc for now)

users then add [library instrumentations](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks) for other instrumentations that they need (hopefully some more of these can be autoinstrumented in our spring boot starter in the future)

looking at other lanaguages, I think it could make sense under Automatic Instrumentation (you pull it in as a dependency, and then the OTel SDK is automatically set up and a few libraries are automatically instrumented).

this would probably mean splitting Automatic instrumentation in two though:

* Java agent
* Spring Boot

potentially we could also add Quarkus here (a popular spring boot alternative) which ships with some automatic instrumentation similar to what we provide for Spring Boot

Reviewer: Thanks for clarification @trask! Let's keep it as it is right now, but consider how we want to have this look like in the future

Reviewer 3: At a minimum the file should be renamed to `spring-boot.md` to match the usual hyphenated naming used elsewhere.

Reviewer: @jeanbisutti please rename the file as suggested

Reviewer: yes, this will be there automatically

Reviewer: you have to move it into `automatic/`

Author: Done. Thank you.

---
title: Spring Boot
linkTitle: Spring Boot

> aliases:
>   - /docs/java/getting_started
>   - /docs/java/springboot
>   - /docs/instrumentation/java/springboot

Reviewer: ```suggestion
```

weight: 30
description: Spring instrumentation for OpenTelemetry Java

> javaInstrumentationVersion: 1.31.0

Reviewer: I think you can inherit this from the _index, @chalin can you provide some guidance here?

Reviewer 3: Yes, let me refactor a few things first before we bring in this PR.

Reviewer 3: @svrnm see:

- #3503

PR #3503 should be merged first, and then this PR rebased and adapted to using the new key name.

Reviewer: PR is merged, so what changes are needed now?

Reviewer 3: See https://github.com/open-telemetry/opentelemetry.io/pull/3377#pullrequestreview-1727954944.

cSpell:ignore: springboot autoconfigure 
---

> You can automatically instrument a [Spring Boot](https://spring.io/projects/spring-boot) application using the [OpenTelemetry Java agent](automatic/_index) that uses byte code instrumentation.

Reviewer: ```suggestion
You can automatically instrument a [Spring Boot](https://spring.io/projects/spring-boot) application using the [OpenTelemetry Java agent](../automatic/) that uses byte code instrumentation.
```

> The OpenTelemetry starter is another way to instrument your Spring Boot application. It is a [Spring Boot starter](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.build-systems.starters).

Reviewer: ```suggestion
The OpenTelemetry Spring Boot Starter is another way to instrument your Spring Boot application. It is a [Spring Boot starter](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.build-systems.starters).
```

I am not 100% sure what the exact name of this should be, but Starter should be capitalized and it should be part of the name that this is for Spring

---
title: Spring Boot
linkTitle: Spring Boot
aliases:
  - /docs/java/getting_started
  - /docs/java/springboot
  - /docs/instrumentation/java/springboot
weight: 30
description: Spring instrumentation for OpenTelemetry Java

> # prettier-ignore

Reviewer: ```suggestion
cSpell:ignore: springboot autoconfigure 
```

this will make sure that cspell stops complaining

---

> You can instrument a Spring Boot application with the [OpenTelemetry Java agent](automatic/_index).
> 
> Another option is to use the OpenTelemetry Spring Boot starter. The rest of this page document it.

Reviewer: ```suggestion
You can automatically instrument a [Spring Boot](https://spring.io/projects/spring-boot) application using the [OpenTelemetry Java agent](automatic/_index), or you can use the OpenTelemetry Spring Boot starter, as outlined below.
```

## Getting Started

OpenTelemetry Spring Starter is a starter package that allow you to setup distributed tracing.

> Check out [opentelemetry-spring-boot-autoconfigure](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/README.md#features) for the full list of supported libraries and features.

Reviewer: As someone who is not involved here, this is confusing, because the link you share brings me to something called "OpenTelemetry Spring Auto-Configuration", which then links to "OpenTelemetry Spring Starter".

Also, people might not know what a "Starter" and what a "Auto-Configuration" is. Do we have anything we can reference here first to clarify what they are going to get here (especially also compared to the agent)

It is compatible with Spring Boot 2.0, Spring Boot 3.0 and Spring native.

Add the dependency given below in your project to enable the OpenTelemetry Spring Starter.

> You have to replace`OPENTELEMETRY_VERSION` with the latest stable [release](https://search.maven.org/search?q=g:io.opentelemetry).

Reviewer: ```suggestion
You have to replace`OPENTELEMETRY_VERSION` with the latest stable [release](https://search.maven.org/search?q=g:io.opentelemetry).
```

If the starter is in sync with other releases we should be able to pull in the latest version like we do it on the other pages.

Author: It is fixes with 125a3823e30186a39d5bca9ec0437f928591b1cc
